urllib3/util/retry.py

```diff
@@ -194,12 +206,51 @@ def get_backoff_time(self):
         backoff_value = self.backoff_factor * (2 ** (consecutive_errors_len - 1))
         return min(self.BACKOFF_MAX, backoff_value)
 
-    def sleep(self):
-        """ Sleep between retry attempts using an exponential backoff.
+    def parse_retry_after(self, retry_after):
+        if retry_after.isdigit():
+            seconds = int(retry_after)
+        else:
+            retry_date_tuple = email.utils.parsedate(retry_after)
+            if retry_date_tuple is None:
+                raise ValueError("Invalid Retry-After header: %s" % retry_after)
+            retry_date = time.mktime(retry_date_tuple)
+            seconds = retry_date - time.time()
+
+        if seconds < 0:
+            seconds = 0
+
+        return seconds
+
+    def get_retry_after(self, response):
+        """ Get the value of Retry-After in seconds. """
+
+        retry_after = response.getheader("Retry-After")
 
-        By default, the backoff factor is 0 and this method will return
-        immediately.
+        if not retry_after:
+            return
+
+        return self.parse_retry_after(retry_after)
+
+    def sleep_for_retry(self, response=None):
```

Comment:
`sleep` tries to sleep with an exponential backoff if `Retry-After` doesn't exist while `sleep_for_retry` doesn't. We need `sleep_for_retry` for redirection which doesn't need to sleep with the backoff.